compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java

```diff
@@ -482,23 +482,39 @@ private BLangConstantValue calculateMod(BLangConstantValue lhs, BLangConstantVal
         return new BLangConstantValue(result, currentConstSymbol.type);
     }
 
+    private Object calculateNegationForInt(BLangConstantValue value) {
+        return -1 * ((Long) (value.value));
+    }
+
+    private Object calculateNegationForFloat(BLangConstantValue value) {
+        return String.valueOf(-1 * Double.parseDouble(String.valueOf(value.value)));
```

Comment:
Why do we need to maintain float as string?

Reply:
This logic was written by Randika, I just put them into separate functions. But from node builder itself when `TokenKind` is `DECIMAL_FLOATING_POINT_LITERAL_TOKEN` that is how it has been handled. Unlike when `TokenKind` is `INTEGER_LITERAL_TOKEN` or  `HEX_INTEGER_LITERAL_TOKEN` where we parse before assigning  to `value` attribute.